compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/SymbolTable.kt

```diff
@@ -285,6 +285,25 @@ open class SymbolTable : ReferenceSymbolTable {
 
     val unboundEnumEntries: Set<IrEnumEntrySymbol> get() = enumEntrySymbolTable.unboundSymbols
 
+    fun declareFieldWithVisibility(
+        startOffset: Int,
+        endOffset: Int,
+        origin: IrDeclarationOrigin,
+        descriptor: PropertyDescriptor,
+        type: IrType,
+        visibility: Visibility,
```

Comment:
Similarly, would be nice to have a default value in `declareField` instead. Also consider specifying the actual argument value for clarity (when it's for example obviously private or public) in as many call sites as possible

Reply:
Done, except for specifying the argument value in the other call sites, since it's actually unused.

This function is strange to begin with: Most arguments are only used in the default value for `fieldFactory`. However, all but two call-sites pass a different value for `fieldFactory`. This applies to pretty much all `declare*` functions in this file.

A better interface would be to have a `declare*` function which just takes the descriptor and factory and a convenience method which produces the default factory from some other arguments with reasonable defaults.

I'd really like to refactor this code, just not in this PR.

Reply:
Thanks! I agree that this needs refactoring and can be postponed.